src/lib/phdi-building-blocks/phdi_building_blocks/utils.py

```diff
@@ -170,10 +174,33 @@ def http_request_with_retry(
             return
     elif request_type == "GET":
         try:
-            response = requests.get(
+            response = http.get(
                 url,
                 headers=headers,
             )
-            return response
         except Exception:
             logging.exception("GET request to " + url + " failed.")
+
+    if (
+        cred_manager
+        and response.status_code == 401
+        and headers.get("Authorization", "").startswith("Bearer ")
+    ):
+        new_access_token = cred_manager.get_access_token().token
```

Comment:
Nice. Independent of the credential manager work, does it make sense to try to fold any other error handling into this util function? Or should we let the caller handle all of that?